test/functional/autocmd/textyankpost_spec.lua

```diff
@@ -0,0 +1,166 @@
+local helpers = require('test.functional.helpers')
+local clear, eval, eq, insert = helpers.clear, helpers.eval, helpers.eq, helpers.insert
+local feed, execute, expect = helpers.feed, helpers.execute, helpers.expect
+
+
+describe('TextYankPost', function()
+  before_each(function()
+    clear()
+
+    -- emulate the clipboard so system clipboard isn't affected
+    execute('let &rtp = "test/functional/fixtures,".&rtp')
+
+    execute('let g:count = 0')
```

Comment:
I would also suggest using `helpers.meths` for manipulating variables: `meths.set_var('count', 0)` in place of `:let`, `meths.get_var('count')` and `meths.get_var('event')` in place of `eval`, so `eval` will not be needed at all.